src/backend/vulkan/src/lib.rs

```diff
@@ -151,12 +194,20 @@ pub fn create(app_name: &str, app_version: u32, layers: &[&str], extensions: &[&
         mem::transmute(lib.GetInstanceProcAddr(instance, name.as_ptr()))
     });
 
-    let mut physical_devices: [vk::PhysicalDevice; 4] = unsafe { mem::zeroed() };
-    let mut num = physical_devices.len() as u32;
-    assert_eq!(vk::SUCCESS, unsafe {
-        inst_pointers.EnumeratePhysicalDevices(instance, &mut num, physical_devices.as_mut_ptr())
-    });
-    let devices = physical_devices[..num as usize].iter()
+    let physical_devices = {
+        let mut num = 0;
+        assert_eq!(vk::SUCCESS, unsafe {
```

Comment:
I'm fine with this code but I don't think it's worth it. Having a maximum bound of the number of devices here helps reducing the code size.